prime-router/src/main/resources/metadata/hl7_mapping/OML_O21/OML_O21-base.yml

```diff
@@ -40,6 +40,7 @@ elements:
     #          https://github.com/CDCgov/prime-reportstream/issues/15500
```

Comment:
I'm having trouble figuring out what adding `RelatedPerson` to the scope actually does for us; as it is, there's no explicit link between the RP and the patient. Writing the scope this way effectively boils down to the logic of include this coverage if `IN1.16` was populated, and don't include it if it wasn't. That doesn't seem right to me either. (For the record, your pseudocode works but the or operator is `or` rather than `||`.)

Looking closer at the `Coverage` resource, I wonder if it makes the most sense to unconditionally map `Coverage.beneficiary` to the patient and use that as the link... that logically makes sense to me, but that's not a mapping that anything is telling us to do.

Reply:
I'm thinking we should keep as it and removing/updating todo because I don't think what is listed in the mapping inventory makes sense. 

The references column for `IN1[Coverage]` states [Coverage.subscriber=Patient[PID]](https://docs.google.com/spreadsheets/d/1cSYLIXVfIjl7ib2DEGxD96LD9-YVJNQtKul5IwzJtts/edit?gid=0#gid=0&range=L31). My understanding of the References column is that it is supposed to represent a required binding. IE How this should read is that we map Coverage back to IN1 if and only if Coverage.subscriber is a reference to the only patient in the bundle. (Please let me know if this is wrong).

So but this only works if IN1-17 is present and 'patient'. So putting anything more restrictive here would be a problem unless we did something radical like just adding a patient reference to Coverage.extension or Coverage.subscriber.extension or Coverage.beneficiary. There's lots of potential for confusion there and all of this is just to prevent if a fhir bundle is sent with irrelevant coverage resources which sounds very unlikely.

Reply:
Looking a little more, `PV1[Coverage]` has `Coverage.beneficiary.reference=Patient[1].id` listed as reference. (This is a really sparse mapping file, only specifying that `PV1-20` maps to `Coverage.type`.) If the implication is that both are supposed to update the same `Coverage`, then creating the `Coverage.beneficiary` link makes sense...

Reply:
Yeah actually I think you're right. Sorry, my brain has been flipping on the references column and I'm overthinking it. Looking at some examples in our code this column is generally treated as something that should be mapped, so adding a reference to the patient on Coverage.beneficiary and then having the mapping here require that sounds correct to me.